kms/api-client/functions_test.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import subprocess
+
+from googleapiclient import discovery
+
+import functions
+
+
+# Your Google Cloud Platform Key Location
+LOCATION = 'global'
+
+# Your Google Cloud Platform KeyRing name
+KEYRING = 'sample-keyring-28'
+
+# Your Google Cloud Platform CryptoKey name
+CRYPTOKEY = 'sample-key-28'
+
+# Your Google Cloud Platform CryptoKeyVersion name
+VERSION = 1
+
+# A member to add to our IAM policy
+MEMBER = 'user:ryanmats@google.com'
+
+# The role we want our new member to have for our IAM policy
+ROLE = 'roles/owner'
+
+
+def test_create_keyring(capsys, cloud_config):
+    functions.create_keyring(cloud_config.project, LOCATION, KEYRING)
+    out, _ = capsys.readouterr()
+    expected = 'Created KeyRing projects/{}/locations/{}/keyRings/{}.'.format(
+        cloud_config.project, LOCATION, KEYRING)
+    assert expected in out
+
+
+def test_create_cryptokey(capsys, cloud_config):
+    functions.create_cryptokey(
+        cloud_config.project, LOCATION, KEYRING, CRYPTOKEY)
+    out, _ = capsys.readouterr()
+    expected = 'Created CryptoKey projects/{}/'.format(cloud_config.project)
+    expected += 'locations/{}/keyRings/{}/'.format(LOCATION, KEYRING)
+    expected += 'cryptoKeys/{}.'.format(CRYPTOKEY)
+    assert expected in out
+
+
+def test_encrypt_decrypt(capsys, cloud_config, tmpdir):
+    # Write to a plaintext file.
+    tmpdir.join("in.txt").write('SampleText')
+
+    # Construct temporary file names.
+    plaintext_file_name = str(tmpdir.join("in.txt"))
+    encrypted_file_name = str(tmpdir.join("out.txt"))
+    decrypted_file_name = str(tmpdir.join("out2.txt"))
+
+    # Encrypt text and then decrypt it.
+    functions.encrypt(
+        cloud_config.project, LOCATION, KEYRING, CRYPTOKEY,
+        plaintext_file_name, encrypted_file_name)
+    functions.decrypt(
+        cloud_config.project, LOCATION, KEYRING, CRYPTOKEY,
+        encrypted_file_name, decrypted_file_name)
+
+    # Make sure the decrypted text matches the original text.
+    decrypted_text = open(decrypted_file_name).read()
```

Comment:
changed (now setting up variables like 'plaintext_file', and then calling encrypt / decrypt with str(plaintext_file) )